mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java

```diff
@@ -1253,4 +1253,81 @@ public PasswordAuthentication getOpenFaasCredentials() {
+                " contain a status, loadbalancer or ingress element.");
+        }
+        ingresses = service.getStatus().getLoadBalancer().getIngress();
+        if (ingresses.size() != 1) {
+            log.debug("There is no or more than one ingress. There should be only one.");
+            return Optional.empty();
+        }
+
+        String ip = ingresses.get(0).getIp();
+        log.debug("Returning the ip '{}' of the Kubernetes service '{}' in the namespace '{}'", ip, name, namespace);
+        return Optional.ofNullable(ip);
+    }
+
+    /**
+     * Requests the list of public ports of a service. It returns the list of ports or an empty list if there are none.
+     *
+     * @param name      the name of the service.
+     * @param namespace the namespace which contains the service.
+     * @return a list of ports or an empty list.
+     * @throws KubernetesResourceException if there is no such service.
+     */
+    public List<Integer> getPublicPortsOfKubernetesService(String name, String namespace) throws KubernetesResourceException {
+        log.debug("Requesting ports of Kubernetes service '{}' in namespace '{}'", name, namespace);
+        LinkedList<Integer> ports = new LinkedList<>();
+        Service service = getServiceOrThrowKubernetesResourceException(name, namespace);
+        if (service.getSpec() == null) {
+            throw new KubernetesResourceException("The Kubernetes service with the name '" + name + "' in the namespace '" + namespace + "' does not contain a spec object");
+        }
+        List<ServicePort> specPorts = service.getSpec().getPorts();
+        if (specPorts != null) {
+            specPorts.forEach(servicePort -> ports.add(servicePort.getPort()));
+        }
+        log.debug("Returning the ports '{}' of the Kubernetes service '{}' in the namespace '{}'", ports, name, namespace);
+        return ports;
+    }
+
+    /**
+     * Requests the service with the given name in the given namespace or {@code null} if there is no such service
+     *
+     * @param name      the name of the service.
+     * @param namespace the namespace which contains the service.
+     * @return the service in the namespace and with the given name or {@code null}.
+     */
+    public Optional<Service> getService(String name, String namespace) {
+        return Optional.ofNullable(kubernetesClient.services().inNamespace(namespace).withName(name).get());
+    }
+
+    /**
+     * Returns the service with the given name in the namespace or throws a {@link KubernetesResourceException}.
+     *
+     * @param name      the name of the service.
+     * @param namespace the namespace which contains the service.
+     * @return the service with the given name in the namespace.
+     * @throws KubernetesResourceException if there is no such service.
+     */
+    private Service getServiceOrThrowKubernetesResourceException(String name, String namespace) throws KubernetesResourceException {
```

Comment:
Maybe rename to `getServiceOrThrow` since the name is long and the thrown exception is already in the `throws` part.